src/bootstrap/tool.rs

```diff
@@ -683,6 +683,50 @@ impl Step for LldWrapper {
     }
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+pub struct RustAnalyzer {
+    pub compiler: Compiler,
+    pub target: TargetSelection,
+}
+
+impl Step for RustAnalyzer {
+    type Output = Option<PathBuf>;
+    const DEFAULT: bool = true;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        let builder = run.builder;
+        run.path("src/tools/rust-analyzer/crates/rust-analyzer").default_condition(
```

Comment:
this is a strange path. normally this would be the top-level path (src/tools/rust-analyzer), not the specific crate. if you want to allow building a single crate instead of the whole workspace, you can use `run.crates("rust-analyzer")`.

(don't remember if it's crate or crates - definitely *not* `all_crates`, that does something different, sorry. trying to get rid of that function so it's less confusing.)